File: src/main/java/mediathek/daten/DownloadStartInfo.java
package mediathek.daten;

public class DownloadStartInfo {
    /**
     * Size of the download list.
     */
    public int total_num_download_list_entries = 0;
    public int total_starts = 0;
    public int num_abos = 0;
    public int num_downloads = 0;
    /**
     * not yet started but initialized.
     */
    public int initialized = 0;
    public int running = 0;
    /**
     * finished without error.
     */
    public int finished = 0;
    /**
     * finished with error.
     */
    public int error = 0;

    /**
     * determine if it has non standard values.
     * @return true if data was changed
     */
    public boolean hasValues() {
        boolean result = false;
        if (total_starts > 0 || num_abos > 0 || num_downloads > 0 || initialized > 0 ||
                running > 0 || finished > 0 || error > 0)
            result = true;

        return result;
    }
}

File: src/main/kotlin/mediathek/daten/DownloadStartInfo.kt
package mediathek.daten

class DownloadStartInfo {
    /**
     * Size of the download list.
     */
    @JvmField
    var total_num_download_list_entries: Int = 0

    @JvmField
    var total_starts: Int = 0

    @JvmField
    var num_abos: Int = 0

    @JvmField
    var num_downloads: Int = 0

    /**
     * not yet started but initialized.
     */
    @JvmField
    var initialized: Int = 0

    @JvmField
    var running: Int = 0

    /**
     * finished without error.
     */
    @JvmField
    var finished: Int = 0

    /**
     * finished with error.
     */
    @JvmField
    var error: Int = 0

    /**
     * determine if it has non standard values.
     * @return true if data was changed
     */
    fun hasValues(): Boolean =
        total_starts > 0 ||
            num_abos > 0 ||
            num_downloads > 0 ||
            initialized > 0 ||
            running > 0 ||
            finished > 0 ||
            error > 0
}
